src/org/team1540/robot2017/commands/SpinupFlywheel.java

```diff
@@ -6,38 +6,54 @@
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 public class SpinupFlywheel extends Command {
-	
-	public SpinupFlywheel() {
-		requires(Robot.shooter);
-	}
-
-	// Called just before this Command runs the first time
-	@Override
-	protected void initialize() {
-		Robot.shooter.setSpeed(Robot.tuning.getShooterFlywheelSpeed());
-	}
-
-	// Called repeatedly when this Command is scheduled to run
-	@Override
-	protected void execute() {
-		
-	}
-
-	// Make this return true when this Command no longer needs to run execute()
-	@Override
-	protected boolean isFinished() {
-		return true;
-	}
-
-	// Called once after isFinished returns true
-	@Override
-	protected void end() {
-	}
-
-	// Called when another command which requires one or more of the same
-	// subsystems is scheduled to run
-	@Override
-	protected void interrupted() {
-	}
-	
+    private double setpoint = 0;
+    private double currentMotorOutput = 0;
+    private double previousError = 0;
+    private double takeBackHalfVariable = 0;
+
+    public SpinupFlywheel() {
+        requires(Robot.shooter);
+    }
+
+    // Called just before this Command runs the first time
+    @Override
+    protected void initialize() {
+        setpoint = Robot.tuning.getShooterFlywheelSpeed();
+    }
+
+    // Called repeatedly when this Command is scheduled to run
+    @Override
+    protected void execute() {
+        double error = setpoint - Robot.shooter.getSpeed();
+        currentMotorOutput += Robot.tuning.getTBHParameter() * error;
+        if (currentMotorOutput > 1) {
+            currentMotorOutput = 1;
+        } else if (currentMotorOutput < 0) {
+            currentMotorOutput = 0;
+        }
+
+        if (Math.signum(error) != Math.signum(previousError)) {
+            currentMotorOutput = takeBackHalfVariable = 0.5 * (currentMotorOutput + takeBackHalfVariable);
+            previousError = error;
```

Comment:
Should this be in the if statement? I think it should be outside, but think about that since I could be wrong.